statistics/handle/handle_hist.go

```diff
@@ -101,31 +103,35 @@ func (h *Handle) SyncWaitStatsLoad(sc *stmtctx.StatementContext) bool {
 	}
 }
 
-// genHistMissingColumns generates hist-missing columns based on neededColumns and statsCache.
-func (h *Handle) genHistMissingColumns(neededColumns []model.TableColumnID) []model.TableColumnID {
+// removeHistLoadedColumns removed having-hist columns based on neededColumns and statsCache.
+func (h *Handle) removeHistLoadedColumns(neededItems []model.TableItemID) []model.TableItemID {
 	statsCache := h.statsCache.Load().(statsCache)
-	missingColumns := make([]model.TableColumnID, 0, len(neededColumns))
-	for _, col := range neededColumns {
-		tbl, ok := statsCache.Get(col.TableID)
+	remainedItems := make([]model.TableItemID, 0, len(neededItems))
+	for _, item := range neededItems {
+		tbl, ok := statsCache.Get(item.TableID)
 		if !ok {
 			continue
 		}
-		colHist, ok := tbl.Columns[col.ColumnID]
+		if item.IsIndex {
+			remainedItems = append(remainedItems, item)
+			continue
+		}
```

Comment:
Should we check the detailed status of index here, wether it's full loaded or not?
We don't need to load index or column if it's full loaded already.

Reply:
we already filter the indices in `collectSyncIndices`, it will discard the index item if the index stats is full load.